### ISSUE TITLE
Integration is using deprecated `DEVICE_CLASS_*` constants

### ISSUE THREAD

**Issue by frenck** [Home-Is-Where-You-Hang-Your-Hack/sensor.goveetemp_bt_hci#49]
This custom integration uses deprecated `DEVICE_CLASS_*` constants in its codebase.

The `DEVICE_CLASS_*` constants have been deprecated and replaced in Home Assistant Core 2021.12 (over a year ago). I would highly suggest updating/migrating this integration to the new enums.

For example, for the device classes supported by the `sensor` platform, there is now a `SensorDeviceClass` enum. So if a sensor previously used the `DEVICE_CLASS_ENERGY` constant, it should now use `SensorDeviceClass.ENERGY`. Other platforms (like `binary_sensor`, and `number`) provide similar enumerations for their supported device classes.

The migration thus only consists of replacing constants with an enumeration member and is, therefore, very low impact and should be fairly straightforward.

If I can help resolve any questions regarding this change or migration, feel free to ask or respond to this issue. I'm happy to help!

Kindest regards,

../Frenck

**Comment by Thrilleratplay** [comment]
@frenck Thank you.  I will try to have this updated in the next few days.

**Comment by Thrilleratplay** [comment]
@frenck  SensorDeviceClass is already being used [here](https://github.com/Home-Is-Where-You-Hang-Your-Hack/sensor.goveetemp_bt_hci/blob/master/custom_components/govee_ble_hci/sensor.py#L149)  but left `DEVICE_CLASS_*` in for backwards compatibility.  Will this be an issue in future Home Assistant releases?

**Comment by frenck** [comment]
To me, the implementation seems incorrect/conflicting. I do not see any evidence of backward compatibility. The only thing I see is the use of deprecated constants and dead code introduced.

As the new enums are imported and not guarded:

https://github.com/Home-Is-Where-You-Hang-Your-Hack/sensor.goveetemp_bt_hci/blob/5e3c7015c5581cf8d26e6e68a11dc4c112a42533/custom_components/govee_ble_hci/sensor.py#L20

That means that backward compatibility is already not maintained. On older version than Home Asistant 2021.12, that import would not succeed.

As for dead code created, this implements the property with the old constant:

https://github.com/Home-Is-Where-You-Hang-Your-Hack/sensor.goveetemp_bt_hci/blob/5e3c7015c5581cf8d26e6e68a11dc4c112a42533/custom_components/govee_ble_hci/sensor.py#L292-L295

However, this specific class uses an `EntityDescription` as shown in the constructor of this class:

https://github.com/Home-Is-Where-You-Hang-Your-Hack/sensor.goveetemp_bt_hci/blob/5e3c7015c5581cf8d26e6e68a11dc4c112a42533/custom_components/govee_ble_hci/sensor.py#L275

Which gets passed the following entity description using the new enums:

https://github.com/Home-Is-Where-You-Hang-Your-Hack/sensor.goveetemp_bt_hci/blob/5e3c7015c5581cf8d26e6e68a11dc4c112a42533/custom_components/govee_ble_hci/sensor.py#L145-L151

However, the entity description properties are not fully used, as the very first property (with the old constant), acts as an override. This causes most of the implementation details in this sensor class regarding `EntityDescription`s to be dead code / unused.

Here is the source property in the main Core codebase:


```py
    @property
    def device_class(self) -> str | None:
        """Return the class of this device, from component DEVICE_CLASSES."""
        if hasattr(self, "_attr_device_class"):
            return self._attr_device_class
        if hasattr(self, "entity_description"):
            return self.entity_description.device_class
        return None

```
Ref: <https://github.com/home-assistant/core/blob/433c0defbe30330dea9f662a895bfce2a0a473eb/homeassistant/helpers/entity.py#L394-L401>

As you can see, if you don't override the `device_class` in your sensor, it will automatically pick up the entity description.

The right solution is to remove the `device_class` property method from your implementation in this case (and that is probably the case for some other property methods in this integration as well).

So, TL;DR, this property can be fully removed and nothing changes:

https://github.com/Home-Is-Where-You-Hang-Your-Hack/sensor.goveetemp_bt_hci/blob/5e3c7015c5581cf8d26e6e68a11dc4c112a42533/custom_components/govee_ble_hci/sensor.py#L292-L295

../Frenck

**Comment by Thrilleratplay** [comment]
@frenck Thank you for that incredibly detailed response.  I know how busy you are and was expecting just a simple "It is dead code.  Just remove it.".  I have gone ahead and released a version with the `device_class` properties removed.

**Comment by frenck** [comment]
> Thank you for that incredibly detailed response

No problem, you are welcome.

Thanks for the adjustments! Awesome 👍 

../Frenck
